[CI-Server] script: improve bash scripts to install required packages

leemgs opened the pull request:
This commit is to update the existing scripts in order to install required
packages in the server. After this PR, a 'mk-build-deps' command of the
scripts install automatically dependent packages (e.g., libgtest-dev
package is required by unit test.)

**Changes proposed in this PR:**

* Version 3:
1. Added 'yes' command for auto 'Enter' key while running add-apt-repository
2. Added dirname statement to find debian/control file anywhere

* Version 2:
1. Introduced mk-build-deps command to install dependent packages automatically
2. Indicated a goal: 1) for CI-server, 2) for CI-system for readability and maintenance

* Version 1:
1. Added libgtest-dev in script file.

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

leemgs commented:
It is required to test the 'debuild' command to inspect a development environment of the CI server (Not CI system) based on Ubuntu. Then, 'pdebuild' command (= isolated build environment with chroot/fakeroot) is enabled as a standard platform package builder of Ubuntu distro.

ohsewon commented on ci/taos/webapp/install-nnstreamer-packages.sh:
```diff
@@ -12,6 +12,9 @@ echo -e "########## Installing base packages for Gstreamer"
 sudo apt -y install gstreamer1.0-tools gstreamer1.0-plugins-base gstreamer1.0-plugins-good
 
+echo -e "########## Installing base packages for a unit test of nnstreamer"
+sudo apt -y install libgtest-dev
```
:100:

myungjoo commented:
Do:
```
nnstreamer$ mk-build-deps -i debian/control
nnstreamer$ apt install nnstreamer-build-deps.....deb
```

leemgs commented:
> ${pkg}# mk-build-deps -i debian/control
> ${pkg}# apt install ${pkg}-build-deps.....deb
> 

Good idea. BTW, It seems that we do not need "apt install ..." because "mk-build-deps ..." command install dependent packages automatically as following:

* Case study:
```bash
invain@mymate:/work/taos/nnstreamer-github.sec.samsung.net$ export LANG=C
invain@mymate:/work/taos/nnstreamer-github.sec.samsung.net$ sudo mk-build-deps --install debian/control
dh_testdir
dh_testroot
dh_prep
dh_testdir
dh_testroot
dh_install
dh_installdocs
dh_installchangelogs
dh_compress
dh_fixperms
dh_installdeb
dh_gencontrol
dh_gencontrol debug symbol wrapper: all non-arch-all packages for this build platform amd64:
dh_gencontrol debug symbol wrapper: packages to act on:
dh_gencontrol debug symbol wrapper: ignored packages:
dh_md5sums
dh_builddeb
dpkg-deb: building package 'nnstreamer-build-deps' in '../nnstreamer-build-deps_2018.6.15_all.deb'.

The package has been created.
Attention, the package has been created in the current directory,
not in ".." as indicated by the message above!
(Reading database ... 436157 files and directories currently installed.)
Preparing to unpack nnstreamer-build-deps_2018.6.15_all.deb ...
Unpacking nnstreamer-build-deps (2018.6.15) over (2018.6.15) ...
Reading package lists... Done
Building dependency tree
Reading state information... Done
Correcting dependencies...Starting pkgProblemResolver with broken count: 0
Starting 2 pkgProblemResolver with broken count: 0
Done
 Done
Starting pkgProblemResolver with broken count: 0
Starting 2 pkgProblemResolver with broken count: 0
Done
The following additional packages will be installed:
  tensorflow-dev tensorflow-lite-dev
The following NEW packages will be installed:
  tensorflow-dev tensorflow-lite-dev
0 upgraded, 2 newly installed, 0 to remove and 335 not upgraded.
1 not fully installed or removed.
Need to get 60.7 MB/61.2 MB of archives.
After this operation, 673 MB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 http://10.113.136.32/download_trbs/newlive/Tizen:/5.0:/TAOS:/UbuntuTools/Ubuntu16.04  tensorflow-dev 1.9.0 [60.7 MB]
Fetched 60.7 MB in 0s (87.3 MB/s)
Selecting previously unselected package tensorflow-lite-dev.
(Reading database ... 436157 files and directories currently installed.)
Preparing to unpack .../tensorflow-lite-dev_1.9.0_amd64.deb ...
Unpacking tensorflow-lite-dev (1.9.0) ...
Selecting previously unselected package tensorflow-dev.
Preparing to unpack .../tensorflow-dev_1.9.0_amd64.deb ...
Unpacking tensorflow-dev (1.9.0) ...
Setting up tensorflow-lite-dev (1.9.0) ...
Setting up tensorflow-dev (1.9.0) ...
Setting up nnstreamer-build-deps (2018.6.15) ...
invain@mymate:/work/taos/nnstreamer-github.sec.samsung.net$ dpkg -l |  grep tensorflow
ii  tensorflow-dev                                1.9.0                                    

```

leemgs commented on ci/taos/webapp/install-nnstreamer-packages.sh:
```diff
@@ -5,26 +5,19 @@
+echo -e "########## for CI-server: Installing base packages to check example apps of NNstreamer"
+# Note that mk-build-deps needs a equivs package.
+sudo apt-get install devscripts equivs > /dev/null
+sudo mk-build-deps --install debian/control > /dev/null
```
From our experiment, dependent packages are automatically installed by --install option.

ohsewon replied:
Should we change the path to use that command.

leemgs commented:
**Changes proposed in this PR:**

* Version 2:
1. Introduced mk-build-deps command to install dependent packages automatically
2. Indicated a goal: 1) for CI-server, 2) for CI-system for readability and maintenance

* Version 1:
1. Added libgtest-dev in script file.

leemgs commented:
> ${pkg}# mk-build-deps -i debian/control

Applied. Thanks.

ohsewon commented on ci/taos/webapp/install-nnstreamer-packages.sh:
```diff
@@ -5,26 +5,19 @@
-echo -e "[DEBUG] Completed...."
+echo -e "########## for CI-server: Installing base packages to check example apps of NNstreamer"
+# Note that mk-build-deps needs a equivs package.
+sudo apt-get install devscripts equivs > /dev/null
```
'apt' is better than 'apt-get' for consistency.

leemgs replied:
Updated.

leemgs commented:
**UPDATE**

**Changes proposed in this PR:**

* Version 3:
1. Added 'yes' command for auto 'Enter' key while running add-apt-repository
2. Added dirname statement to find debian/control file anywhere

* Version 2:
1. Introduced mk-build-deps command to install dependent packages automatically
2. Indicated a goal: 1) for CI-server, 2) for CI-system for readability and maintenance

* Version 1:
1. Added libgtest-dev in script file.
